### ISSUE TITLE
Using `condensed` in HBPackageNameHeaderCell crashes Preferences.app

### ISSUE THREAD

**Issue by Nosskirneh** [hbang/libcephei#24]
Just copied this directly from TypeStatus github to try it out:
```
        <dict>
            <key>cell</key>
            <string>PSGroupCell</string>
            <key>condensed</key>
            <true/>
            <key>headerCellClass</key>
            <string>HBPackageNameHeaderCell</string>
            <key>icon</key>
            <string>icon.png</string>
            <key>packageIdentifier</key>
            <string>ws.hbang.typestatus2</string>
            <key>packageNameOverride</key>
            <string>TypeStatus</string>
        </dict>
```
and it crashes when trying to load my bundle.

Removing
```
            <key>condensed</key>
            <true/>
```
resolves the issue, but why?

I have imported `<CepheiPrefs/HBPackageNameHeaderCell.h>` in my main settings .m file. Using newest version of cephei (1.10).

Also, when this is removed and the Preferences.app does not crash anymore, the opening of my settings bundle lags tremendously the first time visiting it after a restart of Preferences.app because of the header cell. Any ideas?

**Comment by kirb** [comment]
Is there any crash report or exception in the syslog? (Use Console.app or `deviceconsole` or similar from a computer, not on the device itself.)

There will always have to be some delay the first time this is used as libpackageinfo has to parse `/var/lib/dpkg/status` to find the package metadata. This should be looked at sometime to see if any optimisations are possible, though.

**Comment by Nosskirneh** [comment]
Yeah I figured the lagging was because of that. It's a nifty feature, but maybe make it so that the key `packageIdentifier` isn't required - and if it isn't specified, you'd have to specify `packageNameOverride` instead. This would render not having the version information, but I would say that's a good tradeoff for the lag.

The only thing I see at the syslog when I'm entering the settings pane is
```
Apr 15 11:15:35 A-iP6 ReportCrash[13789] <Notice>: MS:Notice: Injecting: com.apple.CrashReporter [ReportCrash] (1348.22)
Apr 15 11:15:35 A-iP6 ReportCrash[13789] <Notice>: MS:Notice: Loading: /Library/MobileSubstrate/DynamicLibraries/CrashReporter.dylib
Apr 15 11:15:36 A-iP6 ReportCrash[13789] <Notice>: MS:Notice: Loading: /Library/MobileSubstrate/DynamicLibraries/RocketBootstrap.dylib
```

I didn't find anything of interest in the [crash log](https://pastebin.com/NXXgFFdF), but since you're more experienced than I, maybe you will :P

**Comment by kirb** [comment]
I think you're looking in the wrong place. If you're on iOS 10, the ASL log which you access via `ondeviceconsole` or `socat` isn't the same as the (new) unified logging system you get over USB via lockdownd, and can view with macOS Sierra's Console.app or [various others](https://theiphonewiki.com/wiki/System_Log). Crash report indicates it's an exception, so there should definitely be a log with the exception reason.

I've been considering some workarounds for the delay - mainly, having it happen on a background thread as soon as CepheiPrefs is loaded. This would work if it's on a secondary list controller like I do, but if it's on the same list controller that's about to appear, it would have to show nothing until it's done. In that case I'd probably recommend using `packageNameOverride` and living with the fact that the version/author will take another moment to show up.

**Comment by kirb** [comment]
Implemented the workaround in 69f9915; explanation in the comments. Unfortunately it seems it’d be too complex to “wait” for it like I was considering above, so if your prefs bundle is the first to load CepheiPrefs and you have the cell on the initial list controller, it’ll still have the same issue. I’m considering moving back to how we worked before, manually parsing output of `dpkg-query`, but for the moment it mostly solves the problem. I’ll also bring it up with ashikase at some point if I remember.

As for your other issue, I’m almost assuming it might be because you don’t have an icon? It does throw an assertion if you’re using condensed but don’t have an icon. I decided I’d reverse that decision in c62cd55.

**Comment by Nosskirneh** [comment]
First of all, sorry for the late answer. Have had some stuff to do.

I did not know there was a difference between ASL and the unified logging system. The culprit of the crash is probably what you guessed - missing icon. I can not verify this, since I've removed all that goes with Cephei after seeing that I only used the header - and that I wrote my own header to get rid of the delay. Thanks for the help nevertheless!
